thrift-logger/src/main/java/com/pinterest/singer/client/AuditLogbackThriftLogger.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2019 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.singer.client;
+
+import com.pinterest.singer.loggingaudit.thrift.LoggingAuditHeaders;
+import com.pinterest.singer.metrics.OpenTsdbMetricConverter;
+import com.pinterest.singer.thrift.LogMessage;
+import com.pinterest.singer.loggingaudit.client.AuditHeadersGenerator;
+
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.LogbackException;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TException;
+import org.apache.thrift.TFieldIdEnum;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.util.Map;
+
+
+/**
+ * AuditLogbackThriftLogger is a class to log to Singer.
+ *
+ * Logback is used under the hood to log to and rotate the underlying
+ * file. The file is written using TFramedTransport as a format, so
+ * it is compatible with Singer's expected format.
+ */
+public class AuditLogbackThriftLogger extends BaseThriftLogger {
```

Comment:
Rename to `Auditable`?

Reply:
The name AuditLogbackThriftLogger is following naming for LogbackThriftLogger. Both are concrete implementations by extending BaseThriftLogger. I feel AuditLogbackThriftLogger is better.

Reply:
Audit is a verb in this case, Auditable is an adjective. Since classnames should be nouns we can use an adjective with it not a verb.

Reply:
will change to AuditableLogbackThriftLogger in updated PR